crawl.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	connmgr "github.com/libp2p/go-libp2p-connmgr"
+
+	ds "github.com/ipfs/go-datastore"
+	dsync "github.com/ipfs/go-datastore/sync"
+	libp2p "github.com/libp2p/go-libp2p"
+	noise "github.com/libp2p/go-libp2p-noise"
+
+	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	host "github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
```

Comment:
I'm going to nitpick

```suggestion
	peer "github.com/libp2p/go-libp2p-core/peer"
```